core/src/main/java/com/datastax/oss/driver/internal/core/context/DefaultNettyOptions.java

```diff
@@ -83,10 +85,16 @@ public DefaultNettyOptions(InternalDriverContext context) {
             .setThreadFactory(safeFactory)
             .setNameFormat(context.getSessionName() + "-timer-%d")
             .build();
+    Duration tickDuration = config.getDuration(DefaultDriverOption.NETTY_TIMER_TICK_DURATION);
+    // JAVA-2264: tick durations on Windows cannot be less than 10 milliseconds,
+    // see https://github.com/netty/netty/issues/356.
+    if (PlatformDependent.isWindows() && tickDuration.toMillis() < 10) {
```

Comment:
I think we may need to make the floor even higher, at least in my limited testing, setting to 10ms seems to prevent the thread from being in a tight loop, but still has high cpu utilization.

I wonder if we should maybe add another option, like `windows-tick-duration-minimum` and set it to 100ms by default.   If a windows user wants to reduce the tick interval, they will have to reduce this value as well.

Having a high resolution timer is really only needed for speculative execution.  I expect most users won't use it on windows.  I might be assuming too much, but I expect Windows to be more likely used in development settings, and deployed servers would run something like linux.

Reply:
ah, this gets messy though, in [JAVA-2056](https://datastax-oss.atlassian.net/browse/JAVA-2056) we adjusted ticks per duration respective to the tick interval.

Not sure what the best course of action here is.   Maybe we could consider switching back to 100ms in general as we may not understand all the implications of using such a high resolution as 1ms, but there are downsides to doing that too.  If we do that, maybe we also add some logic to our spec exec code to warn if the constant threshold is < the tick duration/2 or something like that.